### ISSUE TITLE
ValueError: dtype for the target variable should be int32 or int64 only

### ISSUE THREAD

**Issue by raultomasmora** [pycaret/pycaret#1609]
**Describe the bug**

In regression it is usual to use independent variables in float.
In Pycaret version 2.3.4 I get the following error in regression when activating "Remove Multicollinearity=True", in "False" it does not give the problem.
ERROR: pycaret regression ValueError: dtype for the target variable should be int32 or int64 only

I understand that it should support float and integer values.

FILE: /usr/local/lib/python3.7/dist-packages/pycaret/internal/preprocess.py
rows 2515, 2516

class Fix_multicollinearity(BaseEstimator, TransformerMixin):
    def fit(self, data, y=None):
	...
	if data[self.target_variable].dtype not in ["int32", "int64"]:
            raise ValueError('dtype for the target variable should be int32 or int64 only')
	...

**To Reproduce**

from pycaret.datasets import get_data
dataset = get_data('diamond')

dataset['Price'] =  dataset['Price'].astype('float32')

from pycaret.regression import *
exp_reg101 = setup(data = dataset, target = 'Price', session_id=123, 
                   remove_multicollinearity = True, multicollinearity_threshold = 0.95) 

---------------------------------------------------------------------------

ValueError                                Traceback (most recent call last)
<ipython-input-9-f41871135e22> in <module>()
      1 from pycaret.regression import *
      2 exp_reg101 = setup(data = dataset, target = 'Price', session_id=123, 
----> 3                    remove_multicollinearity = True, multicollinearity_threshold = 0.95) 

7 frames
/usr/local/lib/python3.7/dist-packages/pycaret/internal/preprocess.py in fit(self, data, y)
   2514 
   2515         if data[self.target_variable].dtype not in ["int32", "int64"]:
-> 2516             raise ValueError('dtype for the target variable should be int32 or int64 only')
   2517 
   2518         # global data1

ValueError: dtype for the target variable should be int32 or int64 only

```python
from pycaret.datasets import get_data
dataset = get_data('diamond')

dataset['Price'] =  dataset['Price'].astype('float32')

from pycaret.regression import *
exp_reg101 = setup(data = dataset, target = 'Price', session_id=123, 
                   remove_multicollinearity = True, multicollinearity_threshold = 0.95) 
```

**Expected behavior**

**Additional context**

I understand that it should support float and integer values.

**Versions**
<2.3.4>

</details>

**Comment by moezali1** [comment]
@raultomasmora I just tried this on the `diamond` dataset and it worked for me. I am also on `2.3.4`

```
from pycaret.datasets import get_data
data = get_data('diamond')

from pycaret.regression import *
s = setup(data, target = 'Price', remove_multicollinearity=True, multicollinearity_threshold = 0.95)

lightgbm = create_model('lightgbm')
```

![image](https://user-images.githubusercontent.com/54699234/134575525-d02ac700-fec2-4126-a424-1848ff0be644.png)

![image](https://user-images.githubusercontent.com/54699234/134575533-11547122-e524-46ea-be0f-9ff94a43fca9.png)

**Comment by raultomasmora** [comment]
Try changing dataset['Price'] which is in int, changing it to float
dataset['Price'] =  dataset['Price'].astype('float32')


FULL
from pycaret.datasets import get_data
dataset = get_data('diamond')

dataset['Price'] =  dataset['Price'].astype('float32')

from pycaret.regression import *
exp_reg101 = setup(data = dataset, target = 'Price', session_id=123, 
                   remove_multicollinearity = True, multicollinearity_threshold = 0.95)

**Comment by raultomasmora** [comment]
I have detected the error in other data in my research, but in order to reproduce it, I have approached it with the diamond database.
I work with independent variables in regression as logarithms, so they have many decimals and are float. In version 2.2.1 I didn't get this error and I could use remove_multicollinearity = True without problems.

**Comment by raultomasmora** [comment]
Related Issuse or bug
fix target missing for multicollinearity [#1508](https://github.com/pycaret/pycaret/pull/1508)

**Comment by moezali1** [comment]
@wkuopt Can you look into this? Please..
